### DISCUSSION TITLE
To get the region of data with elevations

### DISCUSSION THREAD

**Discussion by iamzheng00** [pelicanmapping/osgearth#1881] [Support Q&A]
Hi,
I'm currently attempting to get the region of data with elevations. Is there any way to solve this?

Thanks!

**Comment by gwaldron** [comment]
You can go through the `Map` to find all `ElevationLayer` layers, and then call `getExtent` on each one to get its geospatial extent.

**Reply by iamzheng00** [reply]
Glenn,
Thank you for your rapid reply. I've tried the function you suggested, and  I looked into `getExtent`  function, it  contains information about the geo-boundaries of the layer, but I don't know how to use it to get the elevation in that area.
I'm sorry I didn't describe the question clearly. Please allow me describe the situation more specificly  as follows:

The app has loaded elevation data in Local server.(2 kinds of datasource I made a  switch between `TMSElevationlayer ` and `XYZElevationlayer`.) Then I need to get the point cloud (a list of points) of  a rectangular area by user-defined.

To achieve this, my idea is :  First find the tiles in the selected area, and finally convert the tiles to a point cloud.
What I've tried:  I used`map->getProfile()->createTileKey(x, y, lod)`, x,y are correct value in map's srs. Then `ElevationPool::getTile(tilekey,true, tile,&workingSet, progresscallback)` to get a tile. 

Is there any thing I did wrong or not  consider?
I wish to know is there any good way to translate the tile into a point list? Or is there other better/convenient way to get the point cloud data?
-Z

**Reply by gwaldron** [reply]
After calling getTile() you can get the points by calling tile.getHeightField(). You can then use this along with the GeoExtent fo the tile to figure out the coordinates of each heightfield sample.

**Comment by iamzheng00** [comment]
Thank you so much !!!&nbsp; This really helped.My problem has been solved~ Thankyou again!



---Original---
From: ***@***.***&gt;
Date: Fri, Jan 21, 2022 21:50 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [gwaldron/osgearth] To get the region of data with elevations(Discussion #1881)




 
After calling getTile() you can get the points by calling tile.getHeightField(). You can then use this along with the GeoExtent fo the tile to figure out the coordinates of each heightfield sample.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
